packages/dds/map/src/test/mocha/summary.spec.ts

```diff
@@ -0,0 +1,197 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+import { existsSync, readFileSync, writeFileSync } from "node:fs";
+import path from "node:path";
+import {
+	MockContainerRuntimeFactory,
+	MockFluidDataStoreRuntime,
+	MockStorage,
+} from "@fluidframework/test-runtime-utils";
+import { convertSummaryTreeToITree } from "@fluidframework/runtime-utils";
+import { DirectoryFactory, SharedDirectory } from "../..";
+import { assertEquivalentDirectories } from "./directoryEquivalenceUtils";
+
+// Simple filter to avoid tests with a name that would accidentally be parsed as directory traversal or other confusing things.
+const nameCheck = new RegExp(/^[^"/\\]+$/);
+
+assert(__dirname.match(/dist[/\\]test[/\\]mocha$/));
+const snapshotsFolder = path.join(__dirname, `../../../src/test/mocha/snapshots`);
+assert(existsSync(snapshotsFolder));
+
+let currentTestName: string | undefined;
+let currentTestFile: string | undefined;
+
+function useSnapshotDirectory(dirPath: string = "/"): void {
+	const normalizedDir = path.join(snapshotsFolder, dirPath);
+	// Basic sanity check to avoid bugs like accidentally recursively deleting everything under `/` if something went wrong (like dirPath navigated up directories a lot).
+	assert(normalizedDir.startsWith(snapshotsFolder));
+
+	beforeEach(function (): void {
+		currentTestName = this.currentTest?.title ?? assert.fail();
+		currentTestFile = path.join(normalizedDir, `${currentTestName}.json`);
+	});
+
+	afterEach(() => {
+		currentTestFile = undefined;
+		currentTestName = undefined;
+	});
+}
+
+interface TestScenario {
```

Comment:
```suggestion
/**
 * @remarks - This test suite isn't set up to be easily augmented when map's document format changes.
 * `writeCompatible` may want to be changed to enable storing all snapshots over time for a given scenario.
 * See e.g. SharedString tests.
 */
interface TestScenario {
```

I think it's probably fine to put off organization of this until later, though. Ideally we converge on a way to approach this problem so that we don't need to recreate this sort of logic (it's in a few places right now).